Authentication_and_Authorization/hello-security/list-users.php

```diff
@@ -6,22 +6,41 @@
 <body>
 <a href="index.php"> Take me home, country roads ... </a> <br>
 <?php
+
+include "db-connection.php";
+
+function checkIfRoleIsAdmin($email){
+    
+    $conn = openCon();
+
+    $sql = "SELECT * from person where email = :email;";
+    $resultSet = $conn->prepare($sql);
+
+    $resultSet->bindParam(':email', $email);
+
+    $resultSet->execute() or die("Failed to query from DB!");;
+    $firstrow = $resultSet->fetch(PDO::FETCH_ASSOC);
+    
+    $role = $firstrow['role'];
+    return strcmp($role, "admin");
+}
+
 session_start();
+
 if (!isset($_SESSION["email"])) {
     die("Only authenticated users are allowed");
 }
 
-include "db-connection.php";
-$conn = openCon();
-
-$sql = "SELECT * from person where email = '" . $_SESSION["email"] . "';";
-$resultSet = $conn->query($sql) or die("Failed to query from DB!");
-$firstrow = $resultSet->fetch(PDO::FETCH_ASSOC) or die ("User not found.");
+$email= $_SESSION["email"];
+if (checkIfRoleIsAdmin($email)) {
+    die("Only admin has permission to list users.");
```

Comment:
Here you may not want to give so much information. You may want to consider this kind of information sensitive for the security. Said differently - you just revealed what kind of role you need to sign in with in order to get full access to this page.  A simple "No permission" or something similar is enough.